src/Microsoft.OData.Core/UriParser/Parsers/FunctionOverloadResolver.cs

```diff
@@ -217,45 +215,58 @@ internal static bool ResolveOperationFromList(string identifier, IEnumerable<str
                     throw ExceptionUtil.CreateBadRequestError(ODataErrorStrings.RequestUriProcessor_SegmentDoesNotSupportKeyPredicates(identifier));
                 }
 
-                matchingOperation = candidateMatchingOperations.Single();
+                matchingOperation = candidatesMatchingOperations.Count > 0 ? candidatesMatchingOperations[0] : null;
                 return true;
             }

```

Comment:
Move these up.

Reply:
Can't, refactored into a separate function now. Moving both of the exceptions does not improve the code as we need to iterate through all the operations anyway. 

---
In reply to: [370888720](https://github.com/OData/odata.net/pull/1602#discussion_r370888720) [](ancestors = 370888720)